### ISSUE TITLE
Keyword call arity checks cannot be turned off from config

### ISSUE THREAD

**Issue by pgj** [borkdude/clj-kondo#194]
**version**

`v2019.05.20-alpha-SNAPSHOT`

**platform**

macOS

**problem**

Keyword call arity checks are still done even if `skip-args` is set for them.  I use [`org.senatehouse/expect-call`](https://github.com/meredydd/expect-call) in my code, which heavily relies on using keywords in head positions, such as `:more`, `:never`, and `:do`.  The `:more` and `:do` keywords can have more than two arguments because `:more` needs a function name, parameters, and a body, and `:do` can be combined with `:more`.

For example:

```clj
(expect-call (:more select-keys [_ _] 42)
  (is (= [42 42] [(select-keys {} []) (select-keys 1 2)]))
```

`clj-kondo` interprets the use of these keywords wrong, which leads to false positives, but I did not find a way to turn them off.

**repro**

```sh
$ echo "(select-keys (:more 1 2 3 4) [])" | clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
<stdin>:1:14: error: wrong number of args (4) passed to keyword :more
linting took 6ms, errors: 1, warnings: 0
```

however:

```sh
$ echo "(select-keys (select-keys) [])" | clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
linting took 6ms, errors: 0, warnings: 0
```

and:

```sh
$ echo "(select-keys (select-keys) [])" | clj-kondo --lint -
<stdin>:1:14: error: wrong number of args (0) passed to clojure.core/select-keys
linting took 9ms, errors: 1, warnings: 0
```

**expected behavior**

```sh
$ echo "(select-keys (:more 1 2 3 4) [])" | clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
linting took 6ms, errors: 0, warnings: 0
```

**actual behavior**

See **repro**.

**Comment by borkdude** [comment]
@pgj You're right, keyword arity checks should obey that configuration. I'll fix that. Thanks for reporting this. 
I'd still like to know the **platform** to provide you with a binary for testing (Mac if I remember correctly), but you can also build the Dockerfile if you're in a position to do so, when the fix is in.

**Comment by borkdude** [comment]
@pgj This new Mac version should work: https://2820-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2019.05.20-alpha-SNAPSHOT-macos-amd64.zip

**Comment by pgj** [comment]
Yes @borkdude, the platform is macOS.  I set the *platform* field to "N/A" because the reported bug affects every platform.  With Docker, testing the Linux version even should not be a problem, I believe.

**Comment by borkdude** [comment]
I'll update the issue to clarify why I would like to know the platform :-).
If you're able to test this, I'll push to master and update the Docker image.

**Comment by pgj** [comment]
I have tested it @borkdude, it works.  Thank you!

**Comment by borkdude** [comment]
@pgj I updated both the Docker tags here: https://cloud.docker.com/repository/docker/borkdude/clj-kondo/tags

**Comment by borkdude** [comment]
Fixed with b4f743336fd242c2a57aef3ff94700b354c8ffc3

**Comment by pgj** [comment]
Will the Docker image with the `latest` tag be updated as well?

**Comment by borkdude** [comment]
@pgj The latest tag should also have the fix in it. From the next non-snapshot release I plan to do the following:

latest -> latest non-snapshot release 
2019.xx.xx-SNAPSHOT -> latest master commit

So when people use the latest tag, they will always get the latest non-snapshot release. If people want to have the latest master version, the can use a SNAPSHOT tag. Does that make sense?

**Comment by pgj** [comment]
Sure @borkdude, that is what I expected too.

I asked this because I have pulled the `borkdude/clj-kondo:latest` Docker image from Docker Hub for Jenkins after you said you updated the tags, but the bug still persists there.  Taking a quick look at Docker Hub shows me that the image with the `latest` was updated earlier than the fix was pushed to `master`, the one with the `SNAPSHOT` looks fresher (and I have not tried it).

**Comment by borkdude** [comment]
@pgj Locally I see this:

```
Borkdude@michiel ~/Dropbox/dev/clojure/clj-kondo (master) $ echo "(select-keys (:more 1 2 3 4) [])" | docker run borkdude/clj-kondo clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
linting took 14ms, errors: 0, warnings: 0
Borkdude@michiel ~/Dropbox/dev/clojure/clj-kondo (master) $ docker rmi -f borkdude/clj-kondo
Untagged: borkdude/clj-kondo:latest
Untagged: borkdude/clj-kondo@sha256:2647947b93115aa72aa282e51ede3b374af07bba71401aa1fabd88db8ba13f14
Deleted: sha256:9ddd5e10c7d2104226e064e1f246266f1395ed366e829cd44d0368b3a1f30091
Borkdude@michiel ~/Dropbox/dev/clojure/clj-kondo (master) $ echo "(select-keys (:more 1 2 3 4) [])" | docker run borkdude/clj-kondo clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
Unable to find image 'borkdude/clj-kondo:latest' locally
latest: Pulling from borkdude/clj-kondo
e7c96db7181b: Already exists
94ddbf90d2f7: Already exists
046c4c3d1c42: Already exists
3b633a3dae6b: Already exists
4cb06e3121d5: Already exists
e41b19979242: Already exists
Digest: sha256:2647947b93115aa72aa282e51ede3b374af07bba71401aa1fabd88db8ba13f14
Status: Downloaded newer image for borkdude/clj-kondo:latest
linting took 4ms, errors: 0, warnings: 0
```

**Comment by pgj** [comment]
I mean, as far as I remember, the latest non-snapshot version did not have this feature so the bug...?

**Comment by borkdude** [comment]
@pgj The latest tag on Dockerhub now is equal to the snapsnot version, but from the next non-snapshot version on, it will keep pointing to non-snapshot versions only. Before yesterday, there were no images on Dockerhub and there has been no non-snapshot release meanwhile.

**Comment by pgj** [comment]
Yeah, right.

**Comment by pgj** [comment]
Hm, I think something is wrong with the `latest` Docker image.  If you run your example with the `-i` for doing the `run` command in interactive mode (which is the right way if one wants to pass something on the standard input), you will get this:

```sh
$ echo "(select-keys (:more 1 2 3 4) [])" | docker run -i borkdude/clj-kondo clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
<stdin>:1:14: error: wrong number of args (4) passed to keyword :more
linting took 4ms, errors: 1, warnings: 0
```

**Comment by borkdude** [comment]
@pgj I can reproduce that. I'll look into it and fix it. Meanwhile, this seems to work:

```
echo "(select-keys (:more 1 2 3 4) [])" | docker run -i borkdude/clj-kondo:2019.05.20-alpha-SNAPSHOT clj-kondo --lint - --config '{:linters {:invalid-arity {:skip-args [clojure.core/select-keys]}}}'
```

**Comment by borkdude** [comment]
@pgj I updated the latest image now. Can you test again?

**Comment by pgj** [comment]
Now it works perfectly.
